app/rental_routes.py

```diff
@@ -0,0 +1,79 @@
+from flask import Blueprint, jsonify, request
+from app import db
+from app.models.video import Video
+from app.models.customer import Customer
+from app.models.rental import Rental
+
+rentals_bp = Blueprint("rentals_bp", __name__, url_prefix="/rentals")
+
+
+@rentals_bp.route("/check-out", methods=["POST"])
+def check_out():
+    request_body = request.get_json()
+    if "customer_id" not in request_body:
+        response_body = {"details": "Request body must include customer_id."}
+        return jsonify(response_body), 400
+    elif "video_id" not in request_body:
+        response_body = {"details": "Request body must include video_id."}
+        return jsonify(response_body), 400
+
+    video_id = request_body["video_id"]
+    customer_id = request_body["customer_id"]
+    video = Video.query.get(video_id)
+    if not video:
+        return jsonify(None), 404
+    video_inventory = video.total_inventory - len(video.rentals)
+    if video_inventory == 0:
+        response_body = {"message": "Could not perform checkout"}
+        return jsonify(response_body), 400
+
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return jsonify(None), 404
+
+    new_rental = Rental(
+        video_id=request_body["video_id"],
+        customer_id=request_body["customer_id"]
+    )
+
+    db.session.add(new_rental)
+    customer.videos_checked_out = len(customer.rentals)
+    db.session.commit()
+    response_body = new_rental.rental_dict()
+    return jsonify(response_body), 200
+
+
+@rentals_bp.route("/check-in", methods=["POST"])
+def check_in():
+    # TODO: make a helper function(s) to make this check and query. Repeating logic from checkout endpoint. 
+    request_body = request.get_json()
+    if "customer_id" not in request_body:
+        response_body = {"details": "Request body must include customer_id."}
+        return jsonify(response_body), 400
+    elif "video_id" not in request_body:
+        response_body = {"details": "Request body must include video_id."}
+        return jsonify(response_body), 400
+
+    video_id = request_body["video_id"]
+    video = Video.query.get(video_id)
+    customer_id = request_body["customer_id"]
+    customer = Customer.query.get(customer_id)
+
+    # Separate checks beacuse they return different status codes. 
+    if not (video and customer):
+        response_body = {"message": f"No outstanding rentals for customer {customer_id} and video {video_id}"}
+        return jsonify(response_body), 404
```

Comment:
Notice the test for this doesn't check the message. The 404 (not found) message could indicate which of the customer or video wasn't found, not that there are no outstanding rentals.